### PR TITLE
Fix QEMU boot problem

### PR DESCRIPTION
**Pull request by CodeLongAndProsper90**
Implement the fix described by @subinacl and @felixthian in the main source.

### CODE REVIEW

File: init/src/boot_16.cpp [hunk 1]
```diff
@@ -12,6 +12,8 @@
 #include "boot_32.hpp"
 
 #include "gdt_types.hpp"
+void __attribute__ ((noreturn)) rm_main();
+void __attribute__ ((noreturn)) foo(){ rm_main(); }
```
Comment (wichtounet) [inline init/src/boot_16.cpp]: Could you try to add the foo method below rm_main and avoid having to write rm_main twice?
And maybe add a comment as to why this necessary (You can just mention this fix a boot issue even without the reason).

**Review by wichtounet — APPROVED** [review]
Thanks for submitting the fix. 

If you have time, it would be good if you could test a small thing.

File: init/src/boot_16.cpp (hunk 1 above)
Comment (CodeLongAndProsper90) [reply init/src/boot_16.cpp]: > Could you try to add the foo method below rm_main and avoid having to write rm_main twice?

What do you mean?

Also, I've added the docstrings above.

Comment (wichtounet) [reply init/src/boot_16.cpp]: I mean, can you move `void __attribute__ ((noreturn)) foo(){ rm_main(); }` below the definition of rm_main (at the very end of the file) and see if that works?

Comment (CodeLongAndProsper90) [reply init/src/boot_16.cpp]: Nope. Can't get to a shell.

Comment (wichtounet) [reply init/src/boot_16.cpp]: Thanks for testing it!

**Pull request merged**